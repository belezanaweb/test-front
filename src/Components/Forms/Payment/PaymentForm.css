.PaymentForm {
  display: flex;
  flex-direction: column;
  border-radius: 3px;
  background-color: #fff;
  box-shadow: 1px 1px 5px 0 rgba(0, 0, 29, 0.22);
  padding-bottom: 13px;
}

.PaymentForm input {
  display: flex;
  border-radius: 3px;
  height: 45px;
  background-color: #fff;
  border: 1px solid #e7e7e7;
  box-shadow: inset 0 1px 2px 0 rgba(0, 0, 0, 0.2);
  margin: 2px 10px 13px 10px;
}

.PaymentForm label {
  height: 14px;
  color: #ccc;
  font-family: 'Helvetica Neue';
  font-size: 12px;
  font-weight: 700;
  line-height: 14px;
  margin: 12px 0 0px 12px;
}

.formatCol {
  display: flex;
  flex-direction: column;
}

.formatRow {
  display: flex;
}

button {
  margin-top: 20px;
  height: 60px;
  width: 341px;
  border-radius: 3px;
  background-color: #ff6c00;
  border: none;
  box-shadow: inset 0 -3px 0 0 #d45a00, 0 2px 4px 0 rgba(0, 0, 0, 0.25);
  color: #fff;
  font-family: 'Helvetica Neue';
  font-size: 20px;
  font-weight: 700;
  letter-spacing: 0.5px;
  text-align: center;
  cursor: pointer;
}

button:hover {
  background-color: #d45a00;
}

.alert {
  margin-left: 20px;
  color: red;
  font-size: 10px;
}

@media screen and (min-width: 850px) {
  .PaymentForm {
    display: flex;
    flex-direction: row;
    width: 850px;
  }

  .PaymentForm input {
    font-size: 16px;
  }
}

@media screen and (min-width: 1400px) {
  .PaymentForm {
    display: flex;
    flex-direction: row;
    width: 1200px;
  }

  .PaymentForm input {
    font-size: 24px;
  }
}
