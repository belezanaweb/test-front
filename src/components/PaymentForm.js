import React, { Component } from "react";
import { Container } from './styles';

class Form extends Component {

  constructor(props) {
    super(props);
    this.state = {      
      numeroValido: false,
      titularValido: false,
      validadeValido: false,
      cvvValido: false,
    };
  }

  validateForm() {
    let numeroValido = this.state.numeroValido;
    let titularValido = this.state.titularValido;
    let validadeValido = this.state.validadeValido;
    let cvvValido = this.state.cvvValido;
    
    if(!numeroValido && !titularValido && !validadeValido && !cvvValido){
      this.props.disabled(false);
    }else{
      this.props.disabled(true);
    }
  }

  validate = name => event => {
    let erro = false;
    switch (name) {
      case 'numero':
        let regNumero = /\d{4}.\d{4}.\d{4}.\d{4}/;
        if(this.props.numero.length < 16 && !regNumero.test(this.props.numero)){
          erro = true;
          break;
        }else{
          erro = false;
          break;
        }
      case 'titular':
        if(this.props.titular.length < 3){
          erro = true;
          break;
        }else{
          erro = false;          
          break;
        }
      case 'validade':
        let regValidade = /\d{2}[/]\d{4}/;
        if(regValidade.test(this.props.validade)){
          erro = false;
          break;
        }else{
          erro = true;
          break;
        }
      case 'cvv':
        let regCvv = /\d{3}/;
        if(regCvv.test(this.props.cvv)){
          erro = false;
          break;
        }else{
          erro = true;
          break;
        }
      default:
        break;
    }
    this.setState({[name + 'Valido']: erro}, this.validateForm());
    
  }

  render(){
    return(
      <Container>
        <form>
          <div className="row">
            <div className="col">
              <label>Número do cartão:</label>
              <input
                name='numero'
                maxLength='19'
                value={this.props.numero}
                placeholder={'____.____.____.____'}
                onChange={this.props.handleChange}
                onBlur={this.validate('numero')}
                className={ this.state.numeroValido ? 'erro' : undefined }
              />
              <p className="erro">{ this.state.numeroValido && 'Invalid Field' }</p>
            </div>
          </div>

          <div className="row">
            <div className="col">
              <label>Nome do Titular:</label>
              <input
                name='titular'
                maxLength='50'
                value={this.props.titular}
                placeholder={'Como no cartão'}
                onChange={this.props.handleChange}
                onBlur={this.validate('titular')}
                className={ this.state.titularValido ? 'erro' : undefined }
              />
              <p className="erro">{ this.state.titularValido && 'Invalid Field'}</p>
            </div>
          </div>

          <div className="row">
            <div className="col w-160">
              <label>Validade (mês/ano):</label>
              <input
                name='validade'
                maxLength='7'
                value={this.props.validade}
                placeholder={'__/____'}
                onChange={this.props.handleChange}
                onBlur={this.validate('validade')}
                className={ this.state.validadeValido? 'erro' : undefined }
              />
              <p className="erro">{ this.state.validadeValido && 'Invalid Field'}</p>
            </div>
            <div className="col w-145">
              <label>CVV:</label>
              <input
                name='cvv'
                type='text'
                maxLength='4'
                value={this.props.cvv}
                placeholder={'___'}
                onChange={this.props.handleChange}
                onBlur={this.validate('cvv')}
                className={ this.state.cvvValido ? 'erro' : undefined }
              />
              <p className="erro">{ this.state.cvvValido && 'Invalid Field'}</p>
            </div>
          </div>

        </form>
      </Container>
    )
  }
}

export default Form