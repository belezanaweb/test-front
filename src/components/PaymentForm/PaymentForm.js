import { useReducer } from 'react';
import { useCheckoutContext } from '../../contexts';
import { validateForm, maskCreditCardNumber } from './utils';
import Text from '../shared/Text';
import Input from '../shared/Input';
import { initialState, reducer } from './reducer';
import { checkoutTabs } from '../../enums/checkoutTabs';
import styles from './PaymentForm.module.css';

const { CONFIRMATION_TAB } = checkoutTabs;

const PaymentForm = () => {
  const [state, dispatch] = useReducer(reducer, initialState);
  const { setCurrentTab, setPaymentInfo } = useCheckoutContext();

  const handleInputChange = event => {
    const { name, value } = event.target;
    dispatch({ type: name, field: name, value });
  };

  const handleSubmit = async event => {
    event.preventDefault();
    const errors = validateForm(state);
    dispatch({ type: 'updateErrors', errors });

    if (!Object.keys(errors).length) {
      setPaymentInfo({
        ...state,
        cardNumber: maskCreditCardNumber(state.cardNumber),
      });
      setCurrentTab(CONFIRMATION_TAB);
    }
  };

  return (
    <form
      id="payment-form"
      className={styles.Form}
      onSubmit={handleSubmit}
      data-testid="payment-form"
    >
      <Text component="h2" className={styles.ContentTitle} size="xlarge">
        Cartão de crédito
      </Text>
      <Input
        type="text"
        name="cardNumber"
        label="Número"
        placeholder="0000 0000 0000 0000"
        errorMessage={state.errors.cardNumber}
        value={state.cardNumber}
        onChange={handleInputChange}
      />
      <Input
        type="text"
        name="cardHolderName"
        label="Nome do titular do cartão"
        placeholder="Nome impresso no cartão"
        errorMessage={state.errors.cardHolderName}
        value={state.cardHolderName}
        onChange={handleInputChange}
      />
      <div className={styles.ShortCardInformations}>
        <Input
          type="text"
          name="expirationDate"
          label="Data de validade"
          placeholder="MM/AA"
          errorMessage={state.errors.expirationDate}
          value={state.expirationDate}
          onChange={handleInputChange}
        />
        <Input
          type="text"
          name="cvv"
          label="Código CVV:"
          placeholder="000"
          errorMessage={state.errors.cvv}
          value={state.cvv}
          onChange={handleInputChange}
        />
      </div>
    </form>
  );
};

export default PaymentForm;
