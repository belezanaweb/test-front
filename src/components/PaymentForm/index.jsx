import React from 'react'
import { SectionHeader } from '../SectionHeader'
import { SectionContainer, InnerContainer, DateCVVContainer, Input } from './styles'
import { ActionButton } from '../ActionButton'
import { Summary } from '../Summary'
import { useForm } from 'react-hook-form'
import { yupResolver } from '@hookform/resolvers/yup'
import * as yup from 'yup'

const paymentValidation = yup.object().shape({
  cardNumber: yup
    .string()
    .required('Campo inválido. Ex: 1234.1234.1234.1234')
    .min(19, 'Campo inválido. Ex: 1234.1234.1234.1234')
    .test('test-number', 'Campo inválido. Ex: 1234.1234.1234.1234', (value) =>
      /(\d{4}\.){3}/g.test(value)
    ),
  name: yup
    .string()
    .required('Campo inválido')
    .test(
      'test-name',
      'Campo inválido. No mínimo um nome e um sobrenome.',
      (value) => value.split(' ').length >= 2
    ),
  expiryDate: yup
    .string()
    .required('Campo inválido. Ex: 07/2024')
    .min(7, 'Campo inválido. Ex: 07/2024')
    .test('test-date', 'Campo inválido. Cartão vencido.', (value) =>
      /(0?[1-9]|1[0-2])\/(20[2-9][2-9]$)/.test(value)
    ),
  cvv: yup.string().required('Campo inválido').min(3, 'Campo inválido. Ex: 667')
})

export const PaymentForm = ({ handlePaymentInformation }) => {
  const {
    register,
    handleSubmit,
    getValues,
    formState: { errors, isValid }
  } = useForm({
    mode: 'all',
    resolver: yupResolver(paymentValidation)
  })

  const handleFormData = (data) => {
    handlePaymentInformation(data)
  }

  return (
    <>
      <SectionContainer>
        <SectionHeader text={'Cartão de Crédito'} />
        <InnerContainer>
          <form id="payment-form" data-testid="payment-form">
            <label htmlFor="cardNumber">
              <span>Número do Cartão:</span>
              <Input
                id="cardNumber"
                data-testid="card-number-input"
                type="text"
                placeholder="____.____.____.____"
                name="cardNumber"
                {...register('cardNumber')}
                maxLength={19}
                error={!getValues('cardNumber') || errors.cardNumber}
              />
              <p>{errors.cardNumber?.message}</p>
            </label>
            <label htmlFor="name">
              <span>Nome do Titular</span>
              <Input
                id="name"
                data-testid="name-input"
                type="text"
                placeholder="Como no cartão"
                name="name"
                {...register('name')}
                maxLength={50}
                error={!getValues('name') || errors.name}
              />
              <p>{errors.name?.message}</p>
            </label>
            <DateCVVContainer>
              <label htmlFor="expiryDate">
                <span>Validade (mês/ano):</span>
                <Input
                  id="expiryDate"
                  data-testid="expiry-date-input"
                  type="text"
                  placeholder="__/____"
                  name="expiryDate"
                  {...register('expiryDate')}
                  maxLength={7}
                  error={!getValues('expiryDate') || errors.expiryDate}
                />
                <p>{errors.expiryDate?.message}</p>
              </label>
              <label htmlFor="cvv">
                <span>CVV:</span>
                <Input
                  id="cvv"
                  data-testid="cvv-input"
                  type="text"
                  placeholder="___"
                  name="cvv"
                  {...register('cvv')}
                  maxLength={3}
                  error={!getValues('cvv') || errors.cvv}
                />
                <p>{errors.cvv?.message}</p>
              </label>
            </DateCVVContainer>
          </form>
        </InnerContainer>
      </SectionContainer>
      <SectionContainer>
        <Summary />
        <ActionButton
          type="submit"
          handleSubmit={handleSubmit(handleFormData)}
          form="payment-form"
          pathToGo={'/confirmation'}
          actionText={'Finalizar o pedido'}
          disabled={!isValid}
        />
      </SectionContainer>
    </>
  )
}
