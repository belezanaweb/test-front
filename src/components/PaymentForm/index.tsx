import React, { useState } from 'react'
import { Payment } from '../../types/productTypes'
import {
  FieldErrors,
  FieldValidate,
  PaymentValidate
} from '../../util/validations'
import Button from '../Button'
import InfoWrapper from '../InfoWrapper'
import Subtotal, { SubtotalProps } from '../Subtotal'
import TextField from '../TextField'
import Cards from 'react-credit-cards'
import * as S from './styles'
import 'react-credit-cards/es/styles-compiled.css'
import { Redirect } from 'react-router-dom'
import MediaMatch from '../MediaMatch'
import Dropdown from '../Dropdown'

type PaymentFormProps = {
  onSubmit: (value: Payment) => void
} & SubtotalProps

type focusCreditCard = 'name' | 'number' | 'expiry' | 'cvc' | undefined

const PaymentForm = ({
  shippingTotal,
  subTotal,
  total,
  discount,
  onSubmit
}: PaymentFormProps) => {
  const [fieldError, setFieldError] = useState<FieldErrors>({})
  const [focused, setFocused] = useState<focusCreditCard>(undefined)
  const [success, setSuccess] = useState(false)
  const [values, setValues] = useState<Payment>({
    creditCard: '',
    nameInCard: '',
    expirationDate: '',
    cvv: ''
  })

  const handleInput = (field: string, value: string) => {
    setValues((s) => ({ ...s, [field]: value }))
  }

  const handleOnBlur = (value: string, field: keyof Payment) => {
    let objError = fieldError
    const errors = FieldValidate(value, field)
    if (Object.keys(errors).length) {
      setFieldError({ ...fieldError, ...errors })
    } else {
      delete objError[field]
      setFieldError(objError)
    }
  }

  const handleSubmit = async (event: React.FormEvent) => {
    event.preventDefault()

    const errors = PaymentValidate(values)

    if (Object.keys(errors).length) {
      setFieldError(errors)
      return
    }
    onSubmit(values)
    setSuccess(true)
    setFieldError({})
  }

  return (
    <S.CartWrapper onSubmit={handleSubmit}>
      <S.BoxPaymentWrapper>
        <InfoWrapper title="CARTÃO DE CRÉDITO">
          <S.FormWrapper>
            <S.CardWrapper>
              <Cards
                cvc={values.cvv}
                expiry={values.expirationDate}
                focused={focused}
                name={values.nameInCard}
                number={values.creditCard}
              />
            </S.CardWrapper>
            <S.FormDetailsWrapper>
              <S.InputGroup isSideToSide={false}>
                <TextField
                  label={'Número do cartão:'}
                  name={'creditCard'}
                  type={'creditCard'}
                  error={fieldError?.creditCard}
                  onInputChange={(v) =>
                    handleInput(
                      'creditCard',
                      v
                        .split('')
                        .filter((char) => /^[0-9]*$/.test(char))
                        .join('')
                    )
                  }
                  handleOnFocus={() => setFocused('number')}
                  handleOnBlur={() => {
                    setFocused(undefined)
                    handleOnBlur(values.creditCard, 'creditCard')
                  }}
                  placeholder={'____.____.____.____'}
                  data-testid="creditCard"
                  mask={'9999.9999.9999.9999'}
                />

                <TextField
                  label={'Nome do Titular:'}
                  name={'nameInCard'}
                  type={'nameInCard'}
                  error={fieldError?.nameInCard}
                  onInputChange={(v) => handleInput('nameInCard', v)}
                  handleOnFocus={() => setFocused('name')}
                  handleOnBlur={() => {
                    setFocused(undefined)
                    handleOnBlur(values.nameInCard, 'nameInCard')
                  }}
                  placeholder={'Como no cartão'}
                  data-testid="nameInCard"
                  noMask={true}
                />
              </S.InputGroup>
              <S.InputGroup isSideToSide={true}>
                <TextField
                  name={'expirationDate'}
                  type={'expirationDate'}
                  error={fieldError?.expirationDate}
                  onInputChange={(v) =>
                    handleInput(
                      'expirationDate',
                      v
                        .split('')
                        .filter((char) => /^[0-9]*$/.test(char))
                        .join('')
                    )
                  }
                  label={'Validade (mês/ano):'}
                  handleOnFocus={() => setFocused('expiry')}
                  handleOnBlur={() => {
                    setFocused(undefined)
                    handleOnBlur(values.expirationDate, 'expirationDate')
                  }}
                  placeholder={'__/__'}
                  data-testid="expirationDate"
                  mask={'99/99'}
                />

                <TextField
                  name={'cvv'}
                  type={'cvv'}
                  error={fieldError?.cvv}
                  onInputChange={(v) =>
                    handleInput(
                      'cvv',
                      v
                        .split('')
                        .filter((char) => /^[0-9]*$/.test(char))
                        .join('')
                    )
                  }
                  handleOnFocus={() => setFocused('cvc')}
                  handleOnBlur={() => {
                    setFocused(undefined)
                    handleOnBlur(values.cvv, 'cvv')
                  }}
                  label={'CVV:'}
                  placeholder={'___'}
                  data-testid="cvv"
                  mask={'999'}
                />
              </S.InputGroup>
            </S.FormDetailsWrapper>
          </S.FormWrapper>
        </InfoWrapper>
      </S.BoxPaymentWrapper>
      <S.PaymentInfo>
        <MediaMatch greaterThan="medium">
          <S.AcessibilityWrapper>
            <Dropdown />
          </S.AcessibilityWrapper>
        </MediaMatch>
        <Subtotal
          shippingTotal={shippingTotal}
          subTotal={subTotal}
          total={total}
          discount={discount}
        />
        <Button
          disabled={
            !values.creditCard ||
            !values.nameInCard ||
            !values.expirationDate ||
            !values.cvv ||
            Object.keys(fieldError).length !== 0
          }
          type={'submit'}
        >
          FINALIZAR O PEDIDO
        </Button>
      </S.PaymentInfo>
      {success && <Redirect to="/success" />}
    </S.CartWrapper>
  )
}

export default PaymentForm
