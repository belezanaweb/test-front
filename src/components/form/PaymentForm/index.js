import React from 'react'
import { useHistory } from 'react-router-dom'
import { Controller, useForm } from 'react-hook-form'
import Input from '../Input'
import * as S from './styles'
import format from '../../../helpers/utility'
import { useDispatch } from 'react-redux'
import { CartActions } from '../../../store/ducks/cart.ducks'

const PaymentForm = ({ formRef }) => {
  const dispatch = useDispatch()
  const history = useHistory()

  const {
    control,
    handleSubmit,
    formState: { errors }
  } = useForm({})

  const onSubmit = (data) => {
    dispatch(CartActions.getCreditCardData(data))
    history.replace('/sucesso')
  }

  return (
    <S.Form onSubmit={handleSubmit(onSubmit)} data-testid="form-test" ref={formRef}>
      <Controller
        name="cardNumber"
        rules={{ required: true }}
        defaultValue=""
        control={control}
        render={({ field: { onChange, value } }) => (
          <Input
            name="cardNumber"
            type="text"
            label="Número do cartão:"
            placeholder="____.____.____.____"
            control={control}
            onChange={onChange}
            value={format.cardNumber(value)}
            error={errors.cardNumber}
          />
        )}
      />
      <Controller
        name="name"
        rules={{ required: true }}
        defaultValue=""
        control={control}
        render={({ field: { onChange, value } }) => (
          <Input
            name="name"
            label="Nome do titular:"
            placeholder="Como no cartão"
            control={control}
            onChange={onChange}
            value={value}
            error={errors.name}
          />
        )}
      />
      <S.Flex>
        <Controller
          name="expirationDate"
          rules={{ required: true }}
          defaultValue=""
          control={control}
          render={({ field: { onChange, value } }) => (
            <Input
              name="expirationDate"
              type="text"
              label="Validade (mês/ano):"
              placeholder="__/____"
              control={control}
              onChange={onChange}
              value={format.expirationDate(value)}
              error={errors.expirationDate}
            />
          )}
        />
        <Controller
          name="cvvCode"
          rules={{ required: true }}
          defaultValue=""
          control={control}
          render={({ field: { onChange, value } }) => (
            <Input
              name="cvvCode"
              type="text"
              label="CVV:"
              placeholder="___"
              control={control}
              onChange={onChange}
              value={format.cvvCode(value)}
              error={errors.cvvCode}
            />
          )}
        />
      </S.Flex>
    </S.Form>
  )
}

export default PaymentForm
