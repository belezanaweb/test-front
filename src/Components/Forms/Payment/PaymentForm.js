import React, { useContext } from 'react'
import { useForm } from 'react-hook-form'

import TotalList from '../../TotalList/TotalList'
import { payments } from '../../../Store/Actions'
import DataContext from '../../../Store/DataContext'
import './PaymentForm.css'

const PaymentForm = (props) => {
  const { dispatch } = useContext(DataContext)
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = useForm({
    mode: 'onChange'
  })
  const onSubmit = (data) => {
    payments(dispatch, data)
    props.onChangePage('Checkout')
  }
  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <div className="PaymentForm">
        <div className="formatCol">
          <label>Número do cartão:</label>
          {errors.cardNumber && errors.cardNumber.type === 'required' && (
            <span className="alert">This is required</span>
          )}
          {errors.cardNumber && errors.cardNumber.type === 'maxLength' && (
            <span className="alert">Max length exceeded</span>
          )}
          {errors.cardNumber && errors.cardNumber.type === 'minLength' && (
            <span className="alert">Min length missing</span>
          )}
          <input
            type="number"
            id="cardNumber"
            {...register('cardNumber', { required: true, minLength: 12, maxLength: 19 })}
            placeholder="**** **** **** ****"
            style={{ paddingLeft: '2px', fontSize: '18px' }}
          />
        </div>
        <div className="formatCol">
          <label>Nome do titular:</label>
          {errors.cardName && errors.cardName.type === 'maxLength' && (
            <span className="alert">Max length exceeded</span>
          )}
          <input
            id="cardName"
            {...register('cardName', { required: true, maxLength: 40 })}
            style={{ paddingLeft: '2px', fontSize: '18px' }}
          />
        </div>
        <div className="formatRow">
          <div className="formatCol">
            <label>Validade (mês/ano):</label>
            {errors.cardValidate && errors.cardValidate.type === 'maxLength' && (
              <span className="alert">Max length exceeded</span>
            )}
            {errors.cardValidate && errors.cardValidate.type === 'minLength' && (
              <span className="alert">Min length missing</span>
            )}
            <input
              id="cardValidate"
              {...register('cardValidate', { required: true, minLength: 5, maxLength: 5 })}
              style={{ paddingLeft: '2px' }}
            />
          </div>
          <div className="formatCol">
            <label>CVV:</label>
            {errors.CVV && errors.CVV.type === 'maxLength' && (
              <span className="alert">Max length exceeded</span>
            )}
            {errors.CVV && errors.CVV.type === 'minLength' && (
              <span className="alert">Min length missing</span>
            )}
            <input
              type="text"
              {...register('CVV', { required: true, minLength: 3, maxLength: 6 })}
              style={{ paddingLeft: '2px' }}
            />
          </div>
        </div>
      </div>
      <TotalList />
      <button type="submit">FINALIZAR O PEDIDO</button>
    </form>
  )
}

export default PaymentForm
