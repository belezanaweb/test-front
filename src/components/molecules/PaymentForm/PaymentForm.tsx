import * as Styles from './PaymentForm.styles'
import { inputFormatCvv, inputFormatCreditCardNumber, inputFormatValidDate } from '../../../utils/mask'
import Input from '../../atoms/Input/Input'
import { InputHTMLAttributes } from 'react';
import { DeepRequired, FieldErrorsImpl, FieldValues, UseFormRegister } from 'react-hook-form';

export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  register: UseFormRegister<FieldValues>
  errors: FieldErrorsImpl<DeepRequired<FieldValues>>
}
export const PaymentForm: React.FC<InputProps> = ({ register, errors }) => {

  return (
    <Styles.Container>
      <Styles.Form>
        <Input
          data-testid="creditCardNumber"
          id="creditCardNumber"
          label="Número do cartão:"
          maxLength={19}
          placeholder="____.____.____.____"
          patterValidator={/^[0-9]{4}\.[0-9]{4}\.[0-9]{4}\.[0-9]{4}$/i}
          error={errors?.creditCardNumber ? errors.creditCardNumber?.message?.toString() : ''}
          handlerInputValidator={inputFormatCreditCardNumber}
          register={register}
        />

        <Input
          data-testid="creditCardName"
          id="creditCardName"
          label="Nome do Titular:"
          placeholder="Como no cartão"
          maxLength={19}
          patterValidator={/^[A-Za-z\s]+$/i}
          error={errors?.creditCardName ? errors.creditCardName?.message?.toString() : ''}
          register={register}
        />

        <Styles.InputsContainer>
          <Input
            data-testid="creditCardDate"
            id="creditCardDate"
            label="Validade (mês/ano):"
            maxLength={7}
            placeholder="__/____"
            patterValidator={/^(0[1-9]|10|11|12)\/20[0-9]{2}$/i}
            error={errors?.creditCardDate ? errors.creditCardDate?.message?.toString() : ''}
            handlerInputValidator={inputFormatValidDate}
            register={register}
          />

          <Input
            data-testid="creditCardCvv"
            id="creditCardCvv"
            label="CVV:"
            maxLength={3}
            placeholder="___"
            patterValidator={/^[0-9]{3}$/i}
            error={errors?.creditCardCvv ? errors.creditCardCvv?.message?.toString() : ''}
            handlerInputValidator={inputFormatCvv}
            register={register}
          />
        </Styles.InputsContainer>

      </Styles.Form>
    </Styles.Container>
  );

}



export default PaymentForm
