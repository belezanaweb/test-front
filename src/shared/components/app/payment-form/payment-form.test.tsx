import React from 'react';
import { render, screen } from '../../../../../config/test-utils';

import PaymentForm from './index';

describe('<PaymentForm />', () => {
  it('should render component with <form> tag', () => {
    render(<PaymentForm />, {});
    expect(screen.getByRole('form'));
  })

  it('should render component with default props and match snapshot', () => {
    const { container } = render(<PaymentForm />, {});
    expect(container.firstChild).toMatchSnapshot();
  })
})
