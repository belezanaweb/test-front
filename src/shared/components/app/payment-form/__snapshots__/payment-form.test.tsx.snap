// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentForm /> should render component with default props and match snapshot 1`] = `
.c0 {
  margin-bottom: 2.5rem;
}

.c1 {
  font-size: 1.2rem;
  display: block;
  color: #ccc;
  font-weight: 700;
  margin-bottom: 0.5rem;
}

.c2 {
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #e7e7e7;
  box-shadow: inset 0 1px 2px 0 #00000033;
  color: #212121;
  font-size: 1.6rem;
  font-weight: 400;
  outline: none;
  padding: 1.2rem;
  width: 100%;
}

.c2:focus {
  border-color: #a43287;
}

.c2::-webkit-input-placeholder {
  color: #e0e7ee;
  opacity: 1;
}

.c2::-moz-placeholder {
  color: #e0e7ee;
  opacity: 1;
}

.c2:-ms-input-placeholder {
  color: #e0e7ee;
  opacity: 1;
}

.c2::placeholder {
  color: #e0e7ee;
  opacity: 1;
}

.c2:-ms-input-placeholder {
  color: #e0e7ee;
}

.c2::-ms-input-placeholder {
  color: #e0e7ee;
}

.c6 {
  border-radius: 4px;
  border: none;
  cursor: pointer;
  font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif;
  padding: 1rem 0.8rem;
  background: #ff6c00;
  box-shadow: inset 0 -3px 0 0 #d45a00,0 2px 4px 0 #00000040;
  color: #fff;
  text-transform: uppercase;
  font-size: 2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  width: 100%;
  background: #ccc;
  box-shadow: inset 0 -3px 0 0 #afafaf,0 2px 4px 0 #fff;
  color: #afafaf;
}

.c6:hover {
  box-shadow: inset 0 -50px 0 0 #d45a00,0 2px 4px 0 #00000040;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-flow: wrap row;
  -ms-flex-flow: wrap row;
  flex-flow: wrap row;
  width: 100%;
}

.c4 {
  width: 50%;
}

.c5 {
  width: 41.66666666666667%;
  margin-left: 8.333333333333334%;
}

<form
  role="form"
>
  <p
    class="c0"
  >
    <label
      class="c1"
      for="creditCardNumber"
    >
      Número do cartão:
    </label>
    <input
      class="c2"
      id="creditCardNumber"
      name="creditCardNumber"
      placeholder="____-____-____-____"
      type="text"
      value=""
    />
  </p>
  <p
    class="c0"
  >
    <label
      class="c1"
      for="name"
    >
      Nome do Titular:
    </label>
    <input
      class="c2"
      id="name"
      name="name"
      placeholder="Como no cartão"
      type="text"
      value=""
    />
  </p>
  <div
    class="c3"
  >
    <div
      class="c4"
    >
      <p
        class="c0"
      >
        <label
          class="c1"
          for="expires"
        >
          Validade (mês/ano):
        </label>
        <input
          class="c2"
          id="expires"
          name="expires"
          placeholder="__/____"
          type="text"
          value=""
        />
      </p>
    </div>
    <div
      class="c5"
    >
      <p
        class="c0"
      >
        <label
          class="c1"
          for="securityCode"
        >
          CVV:
        </label>
        <input
          class="c2"
          id="securityCode"
          name="securityCode"
          placeholder="___"
          type="text"
          value=""
        />
      </p>
    </div>
  </div>
  <button
    class="c6"
    disabled=""
    type="submit"
  >
    Finalizar o pedido
  </button>
</form>
`;
